gadget/install/install.go

```diff
@@ -169,79 +238,41 @@ func Run(model gadget.Model, gadgetRoot, kernelRoot, bootDevice string, options
 		}
 		logger.Noticef("created new partition %v for structure %v (size %v) %s",
 			part.Node, part, part.Size.IECString(), roleFmt)
-		encrypt := (options.EncryptionType != secboot.EncryptionTypeNone)
-
 		if part.Role == gadget.SystemSave {
 			hasSavePartition = true
 		}
+		if part.Role != "" {
+			devicesForRoles[part.Role] = part.Node
```

Comment:
is this right given that installOnePartition can update it? even if this is correct it probably merits some comment

Reply:
Yes, we want to the device node that corresponds to the raw device, rather then the one that got mapped for LUKS. I've added a comment